File: LeetCode/Python/0014. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strings: List[str]) -> str:
        prefix = []
        shortest = float("inf")
        
        for string in strings:
            shortest = min(shortest, len(string))
        
        for index in range(shortest):
            s = set()
            for string in strings:
                s.add(string[index])
            
            if len(s) == 1:
                prefix.append(s.pop())
            else:
                return "".join(prefix)
            
        return "".join(prefix)

File: LeetCode/Java/0014. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder prefix = new StringBuilder();
        int min = Integer.MAX_VALUE;
        
        for (String str: strs)
            min = Math.min(min, str.length());
            
        for (int i = 0; i < min; i++) {
            HashSet<Character> s = new HashSet<>();
            for (String str: strs)
                s.add(str.charAt(i));
            
            
            if (s.size() == 1)
                prefix.append(s.toArray()[0]);
            else
                return prefix.toString();
        }
        
        return prefix.toString();
    }
}
